align.go

```diff
@@ -205,51 +245,77 @@ func (a *Align) export(lines []string) {
 				}
 			}
 
-			word = applyPadding(word, tempColumn, a.columnCounts[columnNum], j, string(p))
+			padLength := countPadding(word, a.columnCounts[columnNum])
+			// a.padder.Grow(padLength + len(word) + (len(surroundingPad) * 2)) // TODO: might be able to do this once
+
+			paddedWord := applyPadding(a.padder, word, string(surroundingPad), tempColumn, padLength, j)
+
+			a.padder.Reset() // empty the buffer for the next iteration.
+
 			columnNum++
 			tempColumn++
 
 			// Do not add a delimiter to the last field
 			// This also properly aligns the output even if there are lines with a different number of fields
 			if a.filterLen > 0 && a.filter[a.filterLen-1] == columnNum {
-				a.writer.WriteString(word + "\n")
+				a.writer.WriteString(paddedWord + "\n")
 				break
-			} else if columnNum == len(words) {
-				a.writer.WriteString(word + "\n")
+			} else if columnNum == len(paddedWord) {
+				a.writer.WriteString(paddedWord + "\n")
 				break
 			}
-			a.writer.WriteString(word + a.sepOut)
+			a.writer.WriteString(paddedWord + a.sepOut)
 		}
 	}
 	a.writer.Flush()
 }
 
-// pad s based on the supplied PaddingOpts.
-func applyPadding(s string, columnNum, count int, j Justification, pad string) string {
-	padLength := countPadding(s, count)
+// applyPadding rebuilds word by adding padding appropriately based on the
+// desired justification, the overall padding length and the supplied surrounding
+// padding string.
+func applyPadding(padder Padder, original, surroundingPad string, columnNum, padLength int, just Justification) string {
 
-	switch j {
+	// add surrounding pad to beginning of column (except for the 1st column)
+	if len(surroundingPad) > 0 {
+		if columnNum > 0 {
+			padder.WriteString(surroundingPad)
+		}
+	}
+
+	switch just {
 	case JustifyLeft:
-		s = trailingPad(s, padLength)
+		padder.WriteString(original)
+		for i := 0; i < padLength; i++ {
+			padder.WriteByte(padchar)
+		}
 	case JustifyRight:
-		s = leadingPad(s, padLength)
+		for i := 0; i < padLength; i++ {
+			padder.WriteByte(padchar)
+		}
+		padder.WriteString(original)
 	case JustifyCenter:
+		// not much of a point to 'center' justification with such a small padding; default it if <= 2.
 		if padLength > 2 {
-			s = trailingPad(s, padLength/2)
-			s = leadingPad(s, padLength-(padLength/2))
+			for i := 0; i < (padLength - (padLength / 2)); i++ {
```

Comment:
Can this be DRY'd up at all?

Reply:
Yes - I meant to circle back.  Thanks.  Will push soon.